### PR TITLE
RecordException in SqlClient instrumentation

### LINKED ISSUE #1570: SqlClient instrumentation does not record Exception.
Exception is available in DiagnosticSource base instrumentation, so it can do Activity.RecordException, under a option `RecordException` similar to AspNetCore.

### PR DESCRIPTION
**Pull request by mbakalov**
Fixes #1570.

## Changes

Record exceptions as ActivityEvents in SqlClient instrumentation.

* [x] Add unit-tests
* [ ] Changes in public API reviewed - there is a public API change! `SqlClientInstrumentationOptions` new public property `RecordException` (netcore only).
* [x] `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

**Comment by mbakalov** [comment]
This is now ready for review.

Highlights of changes:

* Added `bool RecordException` (default false) to the `SqlClientInstrumentationOptions` public API.
* Added tests for it.
* On netcore we are getting an actual exception so `activity.RecordException()` extension is used but on netfx we just stay it is either `Microsoft.Data.SqlClient.SqlException` or `System.Data.SqlClient.SqlException` based on the event source that EndExecute is coming from.
* CHANGELOG updated

File: src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs [hunk 1]
```diff
@@ -196,6 +205,29 @@ private void OnEndExecute(EventWrittenEventArgs eventData)
...
+            var tagsCollection = new ActivityTagsCollection
+            {
+                // The real exception object is unavailable via the EventSource but we
+                // know that it must be of type "SqlException".
+                {
+                    SemanticConventions.AttributeExceptionType,
+
+                    // The new "Microsoft.Data.SqlClient.EventSource" for sure corresponds to "Microsoft.Data.SqlClient.SqlException".
+                    // The old "Microsoft-AdoNet-SystemData" could be either type, but most likely "System.Data.SqlClient.SqlException".
+                    eventSource == this.mdsEventSource ? MdsExceptionName : AdoNetExceptionName
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs:220]: Given the fact that we cannot get actual exception in .NET Framework, what If we make `RecordException` only available in .NETCore?
In .NET Framework case, we are not adding much value by reporting the  (known) Exceptiontype and the errorcode, as errorcode is already available inside Status.

Comment (mbakalov) [reply src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs:220]: Ah, that's a good idea - thanks! Will do.

Comment (mbakalov) [reply src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs]: Okay updated now, sorry for delay! Also fixed up CHANGELOG and README.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**